test/uk/gov/hmrc/hecstubs/controllers/Testdata.scala

```diff
@@ -0,0 +1,109 @@
+ */
+
+package uk.gov.hmrc.hecstubs.controllers
+
+import play.api.libs.json.Json
+
+object Testdata {
+
+  def badJsonResponse(code: String, message: String) = Json.parse(s"""
+                                                                     |{
+                                                                     |  "failures": [
+                                                                     |    {
+                                                                     |      "code": "$code",
+                                                                     |      "reason": "Submission has not passed validation. $message"
+                                                                     |    }
+                                                                     |  ]
+                                                                     |}
+                                                                     |""".stripMargin)
+
+  def badJsonResponseTwo(code1: String, code2: String, message1: String, message2: String) = Json.parse(
+    s"""
+       |{
+       |  "failures": [
+       |    {
+       |      "code": "$code1",
+       |      "reason": "Submission has not passed validation. $message1"
+       |    },
+       |     {
+       |      "code": "$code2",
+       |      "reason": "Submission has not passed validation. $message2"
+       |    }
+       |  ]
+       |}
+       |""".stripMargin
+  )
+
+  def badJsonResponseThree(
+    code1: String,
+    code2: String,
+    code3: String,
+    message1: String,
+    message2: String,
+    message3: String
+  ) = Json.parse(s"""
+                                            |{
+                                            |  "failures": [
+                                            |    {
+                                            |      "code": "$code1",
+                                            |      "reason": "Submission has not passed validation. $message1"
+                                            |    },
+                                            |     {
+                                            |      "code": "$code2",
+                                            |      "reason": "Submission has not passed validation. $message2"
+                                            |    },
+                                            |    {
+                                            |      "code": "$code3",
+                                            |      "reason": "Submission has not passed validation. $message3"
+                                            |    }
+                                            |  ]
+                                            |}
+                                            |""".stripMargin)
+
+  def badJsonResponseFour(
+    code1: String,
+    code2: String,
+    code3: String,
+    code4: String,
+    message1: String,
+    message2: String,
+    message3: String,
+    message4: String
+  ) = Json.parse(s"""
```

Comment:
you could generalise this to something like this:
```scala
def badJsonResponse(codeWithMessages: (String, String)*) = {
  val errorObjects =
    codeWithMessages.toList
      .map{ case (code, reason) : """..."""  }
      .mkString(",")
  Json.parse("""{ "failures" : [ errorObjects ] } """)
}
```

Reply:
I was planning to do something like this
  def badJsonResponse1(codeWithMessages: (InvalidCode, String)*) = {
    val errorObjects = codeWithMessages.toList
      .map { case (code, reason) =>
        ErrorResult(code, s"Submission has not passed validation. $reason")
      }
    Json.toJson(ErrorResponse(errorObjects))
  }
  Will that be ok?
 I feel more comfortable in case class objects :)

Reply:
that'd work I think - the only danger with that is if we've implemented the writes instance weirdly on `ErrorResult` or `ErrorResponse` we may not catch it with the unit test if we pick up the same format instance in the actual code. At least with explicitly writing out the JSON we get a stronger degree of certainty that the error responses are being written as we expect.

Reply:
updated